flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -830,19 +830,24 @@ class TaskManager(
           case StdOutFileRequest =>
             new File(logFilePath.substring(0, logFilePath.length - 4) + ".out");
         }
-        val fis = new FileInputStream(file);
-        Future {
-          val client: BlobClient = blobService.get.createClient()
-          client.put(fis);
-        }(context.dispatcher)
-          .onComplete {
-            case Success(value) => 
-              sender ! value
-              fis.close()
-            case Failure(e) =>
-              sender ! e
-              fis.close()
+        if (file.exists()) {
+          val fis = new FileInputStream(file);
+          Future {
+            val client: BlobClient = blobService.get.createClient()
+            client.put(fis);
           }(context.dispatcher)
+            .onComplete {
+              case Success(value) =>
+                sender ! value
+                fis.close()
+              case Failure(e) =>
+                sender ! e
+                fis.close()
+            }(context.dispatcher)
+        } else {
+          sender ! new IOException("TaskManager log files are unavailable. " +
+            "Log file does not exist.")
```

Comment:
Maybe we could add the path under which we've looked for the log file to the error message.